don't show web3 modal on cart page init and tab change

gravityblast opened the pull request:
##### Description

When the cart is initialized and when a tab is selected, the `tabChange` method is called.
This method was opening the web3 dialog by default, or waiting to connect to web3 in case it was already opened.
I removed that part so that the dialog is opened only at the final checkout action if needed. 
The final checkout action is already doing the same check, opening the dialog in case the wallet is not connected.
##### Refers/Fixes

Fixes: #9991

nutrina reviewed (approved):
LGTM

thelostone-mc commented:
synced up with @gravityblast. Before merging this we need to ensure 
the web3 popup happens for all 3 
- zksync
- polygon
- mainnet

And a clarification at which step it should happen

chibie commented:
@thelostone-mc @gravityblast it's worth noting that checkout recommendation requires wallet connection upon loading the cart page. does this change break that?

gravityblast commented:
> Before merging this we need to ensure the web3 popup happens for all 3
I updated the code to open the web3 modal for polygon and zksync as well.

> @gravityblast looks like we've got conflicts. once those are resolved -> remove the don't merge PR and let's get this re-reviewed (assuming it's done)

just rebased from master now

thelostone-mc reviewed (approved):
LGTMMM

gdixon reviewed (approved):
Codewise LGTM!! 🚀

thelostone-mc commented:
@gravityblast would you be able to resolve the conflicts?

gravityblast commented:
@thelostone-mc just rebased and force pushed to fix the conflicts
